src/rSharp.cpp

```diff
@@ -398,10 +398,18 @@ SEXP make_char_single_sexp(const char* str) {
 	return mkString(str);
 }
 
+bool isObjectArray(RSharpGenericValue** parameterArray, int i)
+{
+	return parameterArray[i]->type == RSharpValueType::OBJECTARRAY;
+}
+
 void free_params_array(RSharpGenericValue** parameterArray, int size)
 {
 	for (int i = 0; i < size; i++)
 	{
+		if (isObjectArray(parameterArray, i))
+			free_params_array(reinterpret_cast<RSharpGenericValue**>(parameterArray[i]->value), parameterArray[i]->size);

```

Comment:
The reason I needed to add OBJECTARRAY type is because the ```free_params_array``` has to be recursive and process sub-arrays so that the allocated memory can be cleaned up